SPEC/FILES.md

```diff
@@ -759,6 +789,47 @@ A great source of [examples][] can be found in the tests for this API.
 
 The Mutable File System (MFS) is a virtual file system on top of IPFS that exposes a Unix like API over a virtual directory. It enables users to write and read from paths without having to worry about updating the graph. It enables things like [ipfs-blob-store](https://github.com/ipfs/ipfs-blob-store) to exist.
 
+#### `files.chmod`
+
+> Change mode for files and directories
+
+##### `ipfs.files.chmod(path, mode, [options])`
+
+Where:
+
+- `path` is the path to the entry to modify.  It might be:
+  - An existing MFS path to a file or a directory (e.g. `/my-dir/my-file.txt`)
+  - An IPFS path (e.g. `/ipfs/QmWGeRAEgtsHW3ec7U4qW2CyVy7eA2mFRVbk1nb24jFyks`)
+  - A [CID][cid] instance (e.g. `new CID('QmWGeRAEgtsHW3ec7U4qW2CyVy7eA2mFRVbk1nb24jFyks')`)
+- `mode` is the new file mode.  It might be:
+  - A string octal, e.g. `'0755'`
+  - A string modification of the existing mode, e.g. `'+x'`, `'-gw'`, etc
+  - An integer, e.g. the returned value from `parseInt('0755', 8)`
```

Comment:
Or just `0o755`? (I'm not sure if supporting the string version is really necessary given that JS has octal support but it may make for nicer JSON)?

Reply:
Octals-in-strings are supported because the recommended eslint configuration [disallows octal literals](https://eslint.org/docs/rules/no-octal) - our users are likely to be passing them around as strings instead as that's what eslint deems to be the 'correct' way to do it.

Reply:
It disallows octal literals of the form '07...' but I'm pretty sure '0o7...' (ES6) is perfectly fine.

Reply:
Fair point - we can drop the string octal stuff as args in that case.

At the moment the js implementation uses string octals for mode values returned from the various RPC HTTP endpoints for the sake of readability, happy to keep that?